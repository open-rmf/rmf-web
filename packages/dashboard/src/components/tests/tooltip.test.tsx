import React from 'react';
import DashboardTooltip from '../tooltip';
import { Typography } from '@material-ui/core';
import { createShallow } from '@material-ui/core/test-utils';
import { render, screen, fireEvent } from '@testing-library/react';

const mount = createShallow();

const buildWrapper = (title: string, id: string, enabled: boolean) => {
  const root = mount(
    <DashboardTooltip title={title} id={id} enabled={enabled}>
      <Typography variant="h5">Tooltip is enabled</Typography>
    </DashboardTooltip>,
  );
  return root;
};

describe('Dashboard Tooltip', () => {
  it('renders without crashing', () => {
    const root = buildWrapper('test', 'test-tooltip', true);
    expect(root).toMatchSnapshot();
    root.unmount();
  });

  it('renders the tooltips when enabled', () => {
    render(
      <DashboardTooltip title="test" id="test-tooltip" enabled={true}>
        <Typography variant="h5">Tooltip is enabled</Typography>
      </DashboardTooltip>,
    );

    fireEvent.mouseEnter(screen.getByRole('heading'));

    expect(screen.getByTestId('tooltip')).toBeDefined();
  });

  it('does not render the tooltips when disabled', () => {
    render(
      <DashboardTooltip title="test" id="test-tooltip" enabled={false}>
        <Typography variant="h5">Tooltip is disabled</Typography>
      </DashboardTooltip>,
    );

    fireEvent.mouseEnter(screen.getByRole('heading'));

    expect(screen.queryByTestId('tooltip')).toBeNull();
  });
});
