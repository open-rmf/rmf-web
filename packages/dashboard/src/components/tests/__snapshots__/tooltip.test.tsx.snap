// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Tooltip renders without crashing 1`] = `
<div>
  <WithStyles(ForwardRef(Tooltip))
    arrow={true}
    className="makeStyles-tooltipWidth-1"
    data-testid="tooltip"
    id="test-tooltip"
    title="test"
  >
    <WithStyles(ForwardRef(Typography))
      variant="h5"
    >
      Tooltip is enabled
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Tooltip))>
</div>
`;
